front/views/GenreDetailsView.tsx

```diff
@@ -0,0 +1,68 @@
+import { Box, Flex, Heading, useBreakpointValue, ScrollView } from 'native-base';
+import { useQueries, useQuery } from '../Queries';
+import { LoadingView } from '../components/Loading';
+import { RouteProps, useNavigation } from '../Navigation';
+import API from '../API';
+import CardGridCustom from '../components/CardGridCustom';
+import SongCard from '../components/SongCard';
+
+const colorRange = ['#364fc7', '#5c940d', '#c92a2a', '#d6336c', '#20c997'];
+
+type GenreDetailsViewProps = {
+	genreId: number;
+}
+
+const GenreDetailsView = ({ genreId }: RouteProps<GenreDetailsViewProps>) => {
+	const genreQuery = useQuery(API.getGenre(genreId))
+	const songsQuery = useQuery(API.getSongsByGenre(genreId))
+	const screenSize = useBreakpointValue({ base: "small", md: "big" });
+	const isMobileView = screenSize == "small";
+	const navigation = useNavigation();
+
+	if (genreQuery.isError || songsQuery.isError) {
+		navigation.navigate('Error');
+		return <></>;
+	}
+	if (!genreQuery.data || songsQuery.data === undefined) {
+		return <LoadingView />;
+	}
+
+	return (
+		<ScrollView>
+			<Box
+			size={'100%'}
+			height={isMobileView ? 200 : 300}
+			width={'100%'}
+			bg={{
+				linearGradient: {
+					colors: [colorRange[Math.floor(Math.random() * 5)] ?? '#364fc7', 'black'],
+					start: [0, 0],
+					end: [0, 1],
+				},}}
+			/>
+			<Heading ml={3} fontSize={50}>{genreQuery.data.name}</Heading>
+			<Flex
+				flexWrap="wrap"
+				direction={isMobileView ? 'column' : 'row'}
+				justifyContent={['flex-start']}
+				mt={4}
+			>
+				<CardGridCustom
```

Comment:
Use `SongCardGrid` instead

Reply:
I created a card grid custom component so w dont have to actually make a card grid component for each type but only the card component and then specify it to the CustomGrid component